public/components/visualization/embeddable/nlq_vis_embeddable.ts

```diff
@@ -93,12 +98,27 @@ export class NLQVisualizationEmbeddable extends Embeddable<
     return pipeline;
   };
 
+  private handleChange() {
+    let dirty = false;
+
+    // Check if timerange has changed
+    if (!isEqual(this.input.timeRange, this.timeRange)) {
+      this.timeRange = cloneDeep(this.input.timeRange);
+      dirty = true;
+    }
+
+    if (this.handler && dirty) {
```

Comment:
```suggestion
    if (this.updateHandler && dirty) {
```
Should be updateHandler?

Reply:
No, it should be handler, because `this.handler` was created async

Reply:
But it seems the detect has been done if updateHandler already. For someone with little knowledge on this part, it is hard to understand why we detect if this.handler exists when calling updateHandler.

Reply:
I think you're right, it seems we don't need to check `this.handler` here.